core/Test/Tasty/Core.hs

```diff
@@ -65,8 +68,21 @@ data Result = Result
     -- @FAIL@.
   , resultTime :: Time
     -- ^ How long it took to run the test, in seconds.
+  , printResultDetails :: Maybe ResultDetailsPrinter
+    -- ^ Optional detailed result printing action.
+    --
+    -- 'printDesultDetails' is used by providers that need to have fine grained
+    -- control about the printed results. Especially when it comes to colorization.
   }
-  deriving Show
+
+instance Show Result where
+  show result
+    =  "Result {"
+    <> shows (resultOutcome result) ", "
+    <> shows (resultDescription result) ", "
+    <> shows (resultShortDescription result) ", "
+    <> shows (resultTime result) ", "
```

Comment:
You want a trailing "," on this string? If not you could look at `intercalate` to simplify this a bit.

Reply:
The trailing "," is not intend, but it was intend to make it fuse nicely. Looking closer on how to apply `shows` correctly here.